Enhanced Installation Section in README.md

egojoseph opened the pull request:
This PR improves the Installation section in the README.md file by:
- Adding GPU setup instructions for TensorFlow and PyTorch.
- Including troubleshooting tips for Python version issues and missing dependencies.
- Separating OS-specific virtual environment activation commands for clarity.
- Providing links to relevant installation guides for backend frameworks.

These enhancements aim to improve the clarity and usability of the installation instructions for new users.

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

egojoseph committed:
Update README.md

Enhanced installation section with troubleshooting, GPU setup, and OS-specific details.

egojoseph committed:
Update README.md

Enhanced installation section with troubleshooting, GPU setup, and OS-specific details.

Rocketknight1 commented:
cc @stevhliu!

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_35094). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

egojoseph committed:
Merge branch 'main' into Installation-Improvement-ByEgo

egojoseph commented:
Hi @Rocketknight1 and @stevhliu, I noticed that this pull request requires workflow approval and a review to proceed. All checks have passed so far and I have updated the branch to sync with main. Could you kindly review and approve the workflows when convenient? Let me know if further changes are required. Thanks!"

LysandreJik commented:
We want the README.md to stay quite minimal, but those changes could be applied in the installation section in the docs!

egojoseph committed:
Merge branch 'main' into Installation-Improvement-ByEgo

egojoseph committed:
Merge branch 'main' into Installation-Improvement-ByEgo

egojoseph committed:
Update installation.md

Updated installation.md to include virtual environment and GPU setup instructions.

egojoseph committed:
Update installation.md

Updated installation.md to include virtual environment and GPU setup instructions.

egojoseph committed:
Update installation.md

Updated installation.md to include virtual environment, troubleshooting and GPU setup instructions.

egojoseph committed:
Update installation.md

egojoseph committed:
Update installation.md

egojoseph committed:
Update installation.md

egojoseph committed:
Update installation.md

Updated installation.md to include virtual environment, troubleshooting functions and GPU setup instructions.

egojoseph committed:
Update installation.md

Updated installation.md to include virtual environment, troubleshooting functions and GPU setup instructions.

egojoseph committed:
Update installation.md

Updated installation.md to include virtual environment, troubleshooting functions and GPU setup instructions.

egojoseph commented:
Hi @LysandreJik, @Rocketknight1, @stevhliu and team,

Thank you for the earlier feedback. Based on your suggestion, I have migrated the detailed installation instructions from the README.md file to the installation.md section under docs/en.

**Summary of Changes: **
Virtual Environment Setup Instructions:

Steps for creating and activating virtual environments across macOS, Linux and Windows, with additional guidance for potential Windows-specific issues.

**GPU Setup Section: **
Optional instructions for GPU acceleration, including compatibility resources for PyTorch and TensorFlow, along with a command to verify GPU availability.

**Troubleshooting Guide: **
Addressing common installation issues like unsupported Python versions and missing dependencies.
I believe these updates provide clear and practical guidance for users while adhering to the minimalistic approach in the README.md file as suggested. Looking forward to your feedback and approval.

Best regards,
@egojoseph

stevhliu commented on README.md:
```diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
 ## Installation
 
-### With pip
+### 1. With pip
```
Not exactly sequential installation instructions, so no need to number it

```suggestion
### pip
```

egojoseph replied:
Hi @stevhliu,
Thank you for pointing this out! I have updated the subheading to remove the numbering as suggested. Please feel free to review and let me know if any further adjustments are needed.
Best regards,

stevhliu commented on README.md:
```diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
 You should install 🤗 Transformers in a [virtual environment](https://docs.python.org/3/library/venv.html). If you're unfamiliar with Python virtual environments, check out the [user guide](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/).
 
-First, create a virtual environment with the version of Python you're going to use and activate it.
+**a) First, create a virtual environment with the version of Python you're going to use and activate it.**
```
```suggestion
First, create a virtual environment with the version of Python you're going to use and activate it.
```

egojoseph replied:
Hi @stevhliu,
Thanks for catching that! I have removed the "a)" formatting to simplify the sentence as suggested.

stevhliu commented on README.md:
````diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
 
-Then, you will need to install at least one of Flax, PyTorch, or TensorFlow.
-Please refer to [TensorFlow installation page](https://www.tensorflow.org/install/), [PyTorch installation page](https://pytorch.org/get-started/locally/#start-locally) and/or [Flax](https://github.com/google/flax#quick-install) and [Jax](https://github.com/google/jax#installation) installation pages regarding the specific installation command for your platform.
+**For macOS/Linux:**
+```python -m venv env
+source env/bin/activate
+```
+**For Windows:**
````
Add a blank line before and after the code snippet

```suggestion
**macOS/Linux**
```python -m venv env
source env/bin/activate
```

egojoseph replied:
Hi @stevhliu,
Thanks for the suggestion! I have added blank lines before and after the code snippets for improved readability.

stevhliu commented on README.md:
````diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
+env\Scripts\activate
+```
+
+**b) Install a backend framework:**
+To use 🤗 Transformers, you must install at least one of Flax, PyTorch, or TensorFlow. Refer to the official installation guides for platform-specific commands:
````
```suggestion
To use 🤗 Transformers, you must install at least one of Flax, PyTorch, or TensorFlow. Refer to the official installation guides for platform-specific commands:
```

egojoseph replied:
Hi @stevhliu,
Thank you for the feedback! I’ve removed the line "b) Install a backend framework:" as suggested.

stevhliu commented on README.md:
````diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
+```python -m venv env
+source env/bin/activate
+```
+**For Windows:**
````
```suggestion
**Windows**
```

egojoseph replied:
Hi @stevhliu,
Thank you for the suggestion! I have simplified "For Windows:" to "Windows" as recommended.  As well as "For macOS/Linux:" to "macOS/Linux"

stevhliu commented on README.md:
````diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
 When one of those backends has been installed, 🤗 Transformers can be installed using pip as follows:
 
-```bash
+**c) Install 🤗 Transformers:**
````
Not necessary to add this

```suggestion
```

egojoseph replied:
Hi @stevhliu,
Thanks for pointing this out! I have removed the unnecessary heading while keeping the code snippet intact.

stevhliu commented on README.md:
```diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
 If you'd like to play with the examples or need the bleeding edge of the code and can't wait for a new release, you must [install the library from source](https://huggingface.co/docs/transformers/installation#installing-from-source).
 
-### With conda
+**d) Optional: Install from source for the latest updates:**
```
```suggestion
```

egojoseph replied:
Hi @stevhliu,
Thanks for pointing this out! I removed the unnecessary heading as suggested.

stevhliu commented on README.md:
````diff
@@ -247,26 +247,47 @@ The model itself is a regular [Pytorch `nn.Module`](https://pytorch.org/docs/sta
+pip install
+```
+
+### 2. With conda
````
```suggestion
### conda
```

egojoseph replied:
Hi @stevhliu,
Thanks for the suggestion! I have simplified the heading "2. With conda" to "conda" as recommended.

stevhliu commented on README.md:
```diff
@@ -280,6 +301,39 @@ Follow the installation pages of Flax, PyTorch or TensorFlow to see how to insta
 
 > **_NOTE:_**  On Windows, you may be prompted to activate Developer Mode in order to benefit from caching. If this is not an option for you, please let us know in [this issue](https://github.com/huggingface/huggingface_hub/issues/1062).
 
+### 3. GPU Setup (Optional)
```
Let's remove this section to keep it more minimal

egojoseph replied:
Hi @stevhliu,
Thanks for the feedback! I have removed the GPU setup section from the README.md as suggested. I’ll ensure this content is included in installation.md for users needing GPU setup guidance and as recommended by you.

stevhliu commented on docs/source/en/installation.md:
````diff
@@ -34,19 +34,19 @@ You should install 🤗 Transformers in a [virtual environment](https://docs.pyt
 
-```bash
-source .env/bin/activate
+#### For macOS/Linux:
 ```
-Activate Virtual environment on Windows
-
-```bash
-.env/Scripts/activate
+python3 -m venv env
+source env/bin/activate
+```
+#### For window OS:
+```
+python -m venv env
+.\env\Scripts\activate
````
I don't think this section really adds anything new, so I think it can also be removed

egojoseph replied:
Hi @stevhliu,
I have removed the "Create and Activate a Virtual Environment" section as suggested. I understand the concern about redundancy. Would it be helpful to refine this section to emphasize the benefits of virtual environments and include troubleshooting tips? Let me know your thoughts.

stevhliu commented on docs/source/en/installation.md:
```diff
@@ -254,3 +254,46 @@ Once your file is downloaded and locally cached, specify it's local path to load
 
 </Tip>
+
+# Troubleshooting
```
```suggestion
## Troubleshooting

See below for some of the more common installation issues and how to resolve them.
```

egojoseph replied:
Hi @stevhliu,
Thanks for the suggestion! I have updated the "Troubleshooting" section to a second-level heading and added the introductory line as recommended.

stevhliu commented on docs/source/en/installation.md:
````diff
@@ -254,3 +254,46 @@ Once your file is downloaded and locally cached, specify it's local path to load
+
+# Troubleshooting
+
+## a) Error: Python Version Not Supported
+Ensure you are using Python 3.9 or later.
+To check your Python version, run:
+```
+python --version
+```
+## b) Error: Missing Dependencies
+
+Install all required dependencies by running:
+```
+pip install -r requirements.txt
+```
+Ensure you’re in the project directory before executing the command.
+
+## c) Windows-Specific Issues
+
+If you encounter issues on Windows, you may need to activate Developer Mode. 
+
+To enable it, navigate to Windows Settings > For Developers > Developer Mode.
+
+Alternatively, create and activate a virtual environment as shown below:
````
(Add the code fences around the code snippets, I removed it so the changes are easier to see here)

```suggestion
### Unsupported Python version

Ensure you are using Python 3.9 or later. Run the command below to check your Python version.

python --version

### Missing dependencies

Install all required dependencies by running the following command. Ensure you’re in the project directory before executing the command.

pip install -r requirements.txt

### Windows-specific

If you encounter issues on Windows, you may need to activate Developer Mode. Navigate to Windows Settings > For Developers > Developer Mode.

Alternatively, create and activate a virtual environment as shown below.
```

egojoseph replied:
Hi @stevhliu,
Thanks for the detailed feedback! I have updated the troubleshooting section with simplified headings and proper code fences around the code snippets as suggested.

stevhliu commented on docs/source/en/installation.md:
````diff
@@ -254,3 +254,46 @@ Once your file is downloaded and locally cached, specify it's local path to load
+.\env\Scripts\activate
+```
+
+### GPU Setup (Optional)
+
+For GPU acceleration, install the appropriate CUDA drivers for your system:
+
+a) PyTorch CUDA Compatibility: (https://pytorch.org/get-started/locally)  
+b) TensorFlow GPU Support: (https://www.tensorflow.org/install/pip)  
+
+#### Verify GPU availability:
+
+To check if your system detects an NVIDIA GPU, run:
````
You can include this in the "Install with pip" section as just regular text before the line that says `For CPU-support only`. This will flow better so users don't have to jump to the end to check GPU instructions during installation.

```suggestion
For GPU acceleration, install the appropriate CUDA drivers for [PyTorch](https://pytorch.org/get-started/locally) and TensorFlow(https://www.tensorflow.org/install/pip).

Run the command below to check if your system detects an NVIDIA GPU.
```

egojoseph replied:
Hi @stevhliu,
I have integrated the GPU setup instructions into the "Install with pip" section as regular text and removed the standalone "GPU Setup (Optional)" section.

stevhliu reviewed (commented):
Thanks for the suggestions!

egojoseph committed:
Merge branch 'main' into Installation-Improvement-ByEgo

egojoseph committed:
Merge branch 'main' into Installation-Improvement-ByEgo

egojoseph committed:
Update README.md

Removed numbering from README.md.

egojoseph committed:
Update README.md

Removed unnecessary "a)" formatting as per maintainer feedback.

egojoseph committed:
Update README.md

Added blank lines around code snippets for better readability.

egojoseph committed:
Update README.md

Removed the line "b) Install a backend framework:" from README.md as per feedback.

egojoseph committed:
Update README.md

Simplified "For Windows:" to "Windows" in README.md as per feedback as well as "For macOS/Linux:" to "macOS/Linux"

egojoseph committed:
Update README.md

Removed unnecessary heading and retained valid code snippet.

egojoseph committed:
Update README.md

Removed unnecessary heading "d) Optional: Install from source for the latest updates" as per feedback.

egojoseph committed:
Update README.md

Removed "GPU Setup (Optional)" section to align with minimal design feedback.

egojoseph committed:
Update installation.md

Removed "Create and Activate a Virtual Environment" section from installation.md as per feedback.

egojoseph committed:
Update installation.md

Adjusted "Troubleshooting" to a second-level heading and added an introductory line as per feedback.

egojoseph committed:
Update installation.md

Updated troubleshooting section with simplified headings and formatted code blocks as per feedback.

egojoseph committed:
Update installation.md

Integrated GPU setup instructions into the "Install with pip" section for better content flow.

egojoseph commented:
Hi @stevhliu,
I have gone through all the comments and addressed each one. Thanks for your detailed feedback, it really helped improve the documentation! Let me know if there’s anything else to adjust.
 @LysandreJik, @Rocketknight1 and team,

stevhliu commented on README.md:
```diff
@@ -280,6 +300,27 @@ Follow the installation pages of Flax, PyTorch or TensorFlow to see how to insta
 
 > **_NOTE:_**  On Windows, you may be prompted to activate Developer Mode in order to benefit from caching. If this is not an option for you, please let us know in [this issue](https://github.com/huggingface/huggingface_hub/issues/1062).
 
+### Troubleshooting
```
Let's also remove the troubleshooting section to keep it minimal

egojoseph replied:
Hi @stevhliu,
Thank you so much for approving the changes and providing the helpful feedback!
I have removed the Troubleshooting section from README.md as suggested, keeping it minimal, while retaining the detailed instructions in docs/source/en/installation.md.
All tests have passed successfully and the branch is ready with no conflicts. Please let me know if there is anything else you need me to address before merging.

stevhliu reviewed (approved):
Thanks, one more change and then we can merge! :)

egojoseph committed:
Merge branch 'main' into Installation-Improvement-ByEgo

egojoseph committed:
Update README.md

Removed Troubleshooting section from README.md for minimalism as per maintainer feedback.
